gdpr/impl.go

```diff
@@ -194,16 +199,16 @@ const pubRestrictNotAllowed = 0
 const pubRestrictRequireConsent = 1
 const pubRestrictRequireLegitInterest = 2
 
-func (p *permissionsImpl) checkPurpose(consent tcf2.ConsentMetadata, vendor api.Vendor, vendorID uint16, purpose tcf1constants.Purpose) bool {
+func (p *permissionsImpl) checkPurpose(consent tcf2.ConsentMetadata, vendor api.Vendor, vendorID uint16, purpose tcf1constants.Purpose, basicEnforce bool) bool {
 	if purpose == consentconstants.InfoStorageAccess && p.cfg.TCF2.PurposeOneTreatment.Enabled && consent.PurposeOneTreatment() {
 		return p.cfg.TCF2.PurposeOneTreatment.AccessAllowed
 	}
 	if consent.CheckPubRestriction(uint8(purpose), pubRestrictNotAllowed, vendorID) {
 		return false
 	}
 
-	purposeAllowed := vendor.Purpose(purpose) && consent.PurposeAllowed(purpose) && consent.VendorConsent(vendorID)
-	legitInterest := vendor.LegitimateInterest(purpose) && consent.PurposeLITransparency(purpose) && consent.VendorLegitInterest(vendorID)
+	purposeAllowed := consent.PurposeAllowed(purpose) && (basicEnforce || (vendor.Purpose(purpose) && consent.VendorConsent(vendorID)))
+	legitInterest := consent.PurposeLITransparency(purpose) && (basicEnforce || (vendor.LegitimateInterest(purpose) && consent.VendorLegitInterest(vendorID)))
```

Comment:
Should these be:
```
purposeAllowed := consent.PurposeAllowed(purpose) && consent.VendorConsent(vendorID) && (basicEnforce || (vendor.Purpose(purpose)))
legitInterest := consent.PurposeLITransparency(purpose) && consent.VendorLegitInterest(vendorID) && (basicEnforce || vendor.LegitimateInterest(purpose))
```
The basic enforcement requirements indicate if we are enforcing the purpose and the vendor, then the user needs to consent to both. My understanding is we are always enforcing vendor consent; I don't see a config flag to disable it. In that case, my code snippet requires `consent.VendorConsent(vendorID)` return true rather than ORing it with the `baseEnforce` flag.

Reply:
As @bretg pointed out in #1750, we don't want to enforce any of the per-vendor checks and the linked spec will be updated to reflect this. This is good as is.